libraries/AP_MSP/AP_MSP_Telem_Backend.cpp

```diff
@@ -1079,18 +1079,30 @@ MSPCommandResult AP_MSP_Telem_Backend::msp_process_out_rc(sbuf_t *dst)
     float pitch = -rc().rc_channel(rcmap->pitch()-1)->norm_input_dz();
     float yaw = rc().rc_channel(rcmap->yaw()-1)->norm_input_dz();
     float throttle = rc().rc_channel(rcmap->throttle()-1)->norm_input_dz();
+    uint16_t aux1 = rc().rc_channel(4)->get_radio_in();
```

Comment:
I'm unhappy about just using channels 4 through 7 here.

We're already using a mapping to get the first four, I think we should use a mapping for the aux channels too.

Could you experiment with adding an `RC_Channel::AUX_FUNC::MSP_AUX_1` option and using that to find the nominated channel, please?  Note that the user may not have nominated one of them, so you do need to check for nullptr and use some default value on that aux channel.

Reply:
Or preferably, just not send the channel if no channel is set to that aux func. Eg if only msp_aux_1 is set, send only 5 channels.

Reply:
... but possibly filling gaps as required with dummy data.

Reply:
How about sending all the channels all the time? The result from this command usually does not go over a data link, so a bit more bandwidth is not a big problem.

Reply:
Doesn't bother me too much.  We can change it later if we feel like it.

But want ot see that mapping :-)